src/webgpu/api/operation/memory_sync/buffer.spec.ts

```diff
@@ -0,0 +1,261 @@
+export const description = `
+Memory Synchronization Tests for Buffer.
+
+- Test write-after-write on a single buffer. Create one single buffer and initiate it to 0.
+Write a number (say 1) into the buffer via render pass, compute pass, or copy. Write another
+number (say 2) into the same buffer via render pass, compute pass, or copy.
+  - x= 1st write type: {storage buffer in {render, compute}, T2B, B2B}
+  - x= 2nd write type: {storage buffer in {render, compute}, T2B, B2B}
+  - for each write, if render, x= {bundle, non-bundle}
+  - if pass type is the same, x= {single pass, separate passes} (note: render has loose guarantees)
+  - if not single pass, x= writes in {same cmdbuf, separate cmdbufs, separate queues}
+`;
+
+import { pbool, poptions, params } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/framework/util/util.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+const SIZE = 4;
+class BufferSyncTest extends GPUTest {
+  // Create a buffer, and initiate it to zero.
+  createBuffer(): GPUBuffer {
+    const data = new Uint32Array(SIZE / 4);
+    for (let i = 0; i < SIZE / 4; ++i) {
+      data[i] = 0;
+    }
+    const buffer = this.device.createBuffer({
+      mappedAtCreation: true,
+      size: SIZE,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST | GPUBufferUsage.STORAGE,
+    });
+    new Uint8Array(buffer.getMappedRange()).set(data);
+    buffer.unmap();
+    return buffer;
+  }
+
+  createBindGroup(
+    pipeline: GPURenderPipeline | GPUComputePipeline,
+    buffer: GPUBuffer
+  ): GPUBindGroup {
+    return this.device.createBindGroup({
+      layout: pipeline.getBindGroupLayout(0),
+      entries: [{ binding: 0, resource: { buffer } }],
+    });
+  }
+
+  // Create a compute pipeline and write given data into storage buffer.
+  createStorageWriteComputePipeline(value: number): GPUComputePipeline {
+    const wgslCompute = `
+      type Data = [[block]] struct {
+        [[offset 0]] a : i32;
+      };
+
+      [[binding 0, set 0]] var<storage_buffer> data : Data;
+      fn main() -> void {
+	data.a = ${value};
```

Comment:
```suggestion
	  data.a = ${value};
```

Similar issues below.

Reply:
Done